skopt/acquisition.py

```diff
@@ -264,3 +267,119 @@ def gaussian_ei(X, model, y_opt=0.0, xi=0.01, return_grad=False):
         return values, grad
 
     return values
+
+
+def gaussian_entropy_search(X, model, space, n_rep_points=50,
+                            n_trial_points=200, measure="EI", n_samples=500,
+                            random_state=None):
+    """
+    Uses entropy search to optimize the acquisition function.
+
+    1. It approximates the search space by a set of representer points sampled from
+       a suitable measure such as the "probability of improvement" or
+       "expected improvement".
+
+    2. p_min at a representer point X_i is calculated by the Integral
+       p(f(x)) \prod_{j=1}^N_rep \Theta(X_j - X_i) \\
+       Where P(f(x)) is the GP posterior and \Theta is the heaviside step function.
+       This can be used to calculate the base entropy.
+
+    3. Then for each candidate point, the change in the mean and the covariance
+       of each of these representer points can be computed. This in turn
+       can be used to compute the change in p_min for each of these representer
+       points.
+
+    Parameters
+    ----------
+    * `X` [array-like, shape=(n_samples, n_features)]:
+        Values where the acquisition function should be computed.
+
+    * `model` [sklearn estimator that implements predict with ``return_std``]:
+        The fit estimator that approximates the function through the
+        method ``predict``.
+        It should have a ``return_std`` parameter that returns the standard
+        deviation.
+
+    * `space` [Space instance]:
+        This is required to sample representer points.
+
+    * `n_rep_points` [int, default 50]:
+        Number of represter points.
+
+    * `n_trial_points` [int, default 200]:
+        Number of trial points sampled uniformly for each representer point
+        candidate.
+
+    * `n_samples` [int, default 500]:
+        Number of samples to sample from the GP posterior to approximate p_min
+
+    * `random_state` [int, RandomState instance, or None (default)]:
+        For reproducible results.
+    """
+    rng = np.random.RandomState(random_state)
+    n_features = X.shape[1]
+
+    # Step 1: Sample `n_representer_points` from the given nonmeasure.
+    # XXX: Use MCMC sampling to sample from the quadrature measure.
+    n_total = n_rep_points*n_trial_points
+    rep_cands = space.transform(space.rvs(n_samples=n_total, random_state=rng))
+    rep_points = np.zeros((n_rep_points, n_features))
+
+    if measure == "LCB":
+        quad_func = gaussian_lcb
+
+    elif measure in ["EI", "PI"]:
+        if measure == "EI":
+            quad_func = lambda X, model: -gaussian_ei(X, model)
+        else:
+            quad_func = lambda X, model: -gaussian_pi(X, model)
+
+    for i, startptr in enumerate(range(0, X.shape[0], n_total)):
```

Comment:
do you need the loop? you can evaluate `quad_func` at all `rep_cands`, then reshape by `(n_rep_points, n_trial_points)` and take the argmin by specifying the axis.